hilti/include/rt/types/set.h

```diff
@@ -63,36 +112,75 @@ class Set : protected std::set<T>, public hilti::rt::detail::iterator::Controlle
     Set& operator=(const Set&) = default;
     Set& operator=(Set&&) noexcept = default;
 
-    /** Returns true if a specific element is part of the set. */
-    bool contains(const T& t) { return this->find(t) != this->end(); }
-
-    // Methods of `std::set`.
-    using V::begin;
-    using V::cbegin;
-    using V::cend;
-    using V::end;
+    /** Checsk whether an element is in the set.
+     *
+     * @param `t` the element to check for
+     * @return `true` if the element is part of the set. */
+    bool contains(const T& t) { return this->count(t); }
+
+    auto begin() const { return iterator(static_cast<const V&>(*this).begin(), empty() ? nullptr : _control); }
+
+    auto end() const { return iterator(static_cast<const V&>(*this).end(), empty() ? nullptr : _control); }
+
+    /** Removes an element from the set.
+     *
+     * This function invalidates all iterators into the set.
```

Comment:
That's a comment worth reiterating more generally in the type's doc header: All modifying operations invalidate outstanding iterators.

Reply:
I added something along those lines to the type doc. I also fixed the signature of `contains` which is not a mutating operation.